src/alire/alire-dependencies-states.ads

```diff
@@ -202,6 +211,13 @@ private
       Transitivity : Transitivities := Unknown;
    end record;
 
+   ---------
+   -- "=" --
+   ---------
+
+   overriding function "=" (L, R : State) return Boolean
+   is (L.Image = R.Image);
```

Comment:
Converting to String and the comparing the String sounds like a not very efficient way of comparison.
This should probably be flagged as an area for improvement.

Reply:
Right. I have lost too much time trying to ascertain what's happening there before resorting to this. I'll flag it with a TODO.